modules/atomic-widgets/widgets/atomic-heading.php

```diff
@@ -44,6 +54,27 @@ protected function render() {
 		);
 	}
 
+	public function get_link_template( array $settings ): string {
+		$tag = $settings['tag'];
+		$title = $settings['title'];
+		$attrs = array_filter( [
+			'class' => $settings['classes'] ?? '',
+		] );
+
+		$link_attrs = [
+			'href' => esc_url( $settings['link']['href'] ) ?? '',
```

Comment:
Since `href` is a Url_Prop_Type - it would be sanitized as part of my [PR](https://github.com/elementor/elementor/pull/29409) so it would probably be an overkill to esc

Reply:
I would prefer if we escape it during render as well. There could be other processes that interfere with the data after it was saved to the DB.